src/go/k8s/pkg/resources/certmanager/pki.go

```diff
@@ -105,43 +105,46 @@ func (r *PkiReconciler) Ensure(ctx context.Context) error {
 	if kafkaListeners := kafkaAPIListeners(r.pandaCluster); len(kafkaListeners) > 0 {
 		toApplyAPI, err := r.prepareAPI(ctx, kafkaAPI, RedpandaNodeCert, []string{OperatorClientCert, UserClientCert, AdminClientCert}, kafkaListeners, &keystoreKey)
 		if err != nil {
-			return err
+			return false, err
 		}
 		toApply = append(toApply, toApplyAPI...)
 	}
 
 	if adminListeners := adminAPIListeners(r.pandaCluster); len(adminListeners) > 0 {
 		toApplyAPI, err := r.prepareAPI(ctx, adminAPI, adminAPINodeCert, []string{adminAPIClientCert}, adminListeners, &keystoreKey)
 		if err != nil {
-			return err
+			return false, err
 		}
 		toApply = append(toApply, toApplyAPI...)
 	}
 
 	if pandaProxyListeners := pandaProxyAPIListeners(r.pandaCluster); len(pandaProxyListeners) > 0 {
 		toApplyAPI, err := r.prepareAPI(ctx, pandaproxyAPI, pandaproxyAPINodeCert, []string{pandaproxyAPIClientCert}, pandaProxyListeners, &keystoreKey)
 		if err != nil {
-			return err
+			return false, err
 		}
 		toApply = append(toApply, toApplyAPI...)
 	}
 
 	if schemaRegistryListeners := schemaRegistryAPIListeners(r.pandaCluster); len(schemaRegistryListeners) > 0 {
 		toApplyAPI, err := r.prepareAPI(ctx, schemaRegistryAPI, schemaRegistryAPINodeCert, []string{schemaRegistryAPIClientCert}, schemaRegistryListeners, &keystoreKey)
 		if err != nil {
-			return err
+			return false, err
 		}
 		toApply = append(toApply, toApplyAPI...)
 	}
+	allApplied := true
 
 	for _, res := range toApply {
-		err := res.Ensure(ctx)
+		applied, err := res.Ensure(ctx)
 		if err != nil {
 			r.logger.Error(err, "Failed to reconcile pki")
+			return false, err
 		}
+		allApplied = allApplied && applied
 	}
 
-	return nil
+	return allApplied, nil
```

Comment:
I'm not sure I fully understand the meaning of the bool that's being returned. Should it only be true if the whole operation succeed? Or should it be true if at least something was changed?

In this case, let's say kafkaAPI certs were successfully created, but PandaProxy certs failed, what should happen 🤔 ?

Reply:
The way I understand it, it should return `true` if it updated something.